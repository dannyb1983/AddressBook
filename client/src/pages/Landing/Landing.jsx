//modules
import React, { useEffect, useState } from "react";
import { Container, Row, Col } from "react-bootstrap";
import { v4 as uuidv4 } from "uuid";

// COMPONENTS
import UserTable from "../../components/UserTable";

const Landing = ({
  people,
  sortPeople,
  setContactToEdit,
  sortType,
  sortTypeToggle,
}) => {
  const [searchInput, setSearchInput] = useState("");
  // Fetch users
  useEffect(() => {
    if (people) {
      sortPeople(people, sortType);
    } else {
      fetch("https://randomuser.me/api/?results=15")
        .then((response) => response.json())
        .then((res) => {
          //give each contact a unique id
          res.results.forEach((el) => (el.login.uuid = uuidv4()));
          //sort results alphabetically
          sortPeople(res.results, sortType);
        })
        .catch((err) => {
          console.log("Error fetching users: ", err);
        });
    }
  }, [sortType]);

  return (
    <Container>
      <Row>
        <Col md={12}>
          {/* Toglle first and last name sort */}
          <h1 onClick={sortTypeToggle}>Contacts</h1>
        </Col>
        <Col md={12}>
          <div style={{ textAlign: "center" }}>
            <input
              className="searchBar"
              onChange={(e) => {
                setSearchInput(e.target.value);
              }}
              style={{
                width: "40%",
                borderRadius: "30px",
                paddingLeft: "20px",
              }}
              type="text"
              placeholder="Search"
            />
          </div>
        </Col>
      </Row>
      {people.length === 0 ? (
        <h4 style={{ textAlign: "center", padding: "50px" }}>
          Add a new contact to get started!
        </h4>
      ) : null}
      <Row>
        <Col md={12} className="mt-5">
          <UserTable
            people={people}
            setContactToEdit={setContactToEdit}
            searchInput={searchInput}
          />
        </Col>
      </Row>
    </Container>
  );
};

export default Landing;
