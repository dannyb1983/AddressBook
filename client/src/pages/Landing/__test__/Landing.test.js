//modules
import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { contact, contacts } from '../../../../../test/jest/__mocks__/testData';
//components
import Landing from '../Landing';

describe('Landing (Unit)', () => {
  afterEach(() => cleanup());
  const renderLanding = () =>
    render(
      <MemoryRouter initialEntries={['/']}>
        <Landing
          people={contacts}
          sortPeople={() => contacts}
          setContactToEdit={contact}
          sortType={'first'}
          sortTypeToggle={() => {
            'first';
          }}
        />
      </MemoryRouter>
    );
    const renderLandingNoPeople = () =>
    render(
      <MemoryRouter initialEntries={['/']}>
        <Landing
          people={[]}
          sortPeople={() => contacts}
          setContactToEdit={contact}
          sortType={'first'}
          sortTypeToggle={() => {
            'first';
          }}
        />
      </MemoryRouter>
    );
  const renderEmptyLanding = () =>
    render(
      <MemoryRouter>
        <Landing />
      </MemoryRouter>
    );

  describe('render Landing()', () => {
    const { asFragment, container } = renderLanding();

    it('Landing render should match snapshot', () => {
      renderLanding();
      expect(asFragment()).toMatchSnapshot();
    });
    
    it('Landing should render the search bar', () => {
      renderLanding();
      expect(screen.getByRole('textbox')).toBeInTheDocument()
    });
    
    
    it('Should render a message if there are no contacts', () => {
      renderLandingNoPeople()
      expect(screen.getByText('Add a new contact to get started!')).toBeInTheDocument()
    });
    
    it('Should not render the "add new contact" message when there are contacts', () => {
      renderLanding()
      const h4 = screen.queryByText('Add a new contact to get started!')
      expect(h4).not.toBeInTheDocument()
    });
    
  });
});
