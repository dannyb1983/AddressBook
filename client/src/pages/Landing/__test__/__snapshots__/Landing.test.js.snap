// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing (Unit) render Landing() Landing render should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-md-12"
      >
        <h1>
          Contacts
        </h1>
      </div>
      <div
        class="col-md-12"
      >
        <div
          style="text-align: center;"
        >
          <input
            class="searchBar"
            placeholder="Search"
            style="width: 40%; border-radius: 30px; padding-left: 20px;"
            type="text"
          />
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="mt-5 col-md-12"
      >
        <ol
          class="list-group"
        >
          <div
            class="list-group-item"
          >
            <div
              class="my-3 my-auto flex justify-content-center container"
            >
              <div
                class="row"
              >
                <div
                  class="mx-auto text-center col-md-6"
                >
                  <div
                    class="ms-2 me-auto"
                  >
                    <div
                      class="fw-bold"
                    >
                      <a
                        href="/details"
                        style="text-decoration: inherit; padding-right: 20px;"
                      >
                         
                        <div
                          class="my-auto"
                        >
                          John Smith 
                        </div>
                      </a>
                    </div>
                  </div>
                </div>
                <div
                  class="my-auto mx-auto text-center py-2 px-1 col-md-6"
                >
                  <img
                    alt="user_picture"
                    class="rounded-circle"
                    height="48px"
                    src="https://randomuser.me/api/portraits/thumb/men/33.jpg"
                    width="48px"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="list-group-item"
          >
            <div
              class="my-3 my-auto flex justify-content-center container"
            >
              <div
                class="row"
              >
                <div
                  class="mx-auto text-center col-md-6"
                >
                  <div
                    class="ms-2 me-auto"
                  >
                    <div
                      class="fw-bold"
                    >
                      <a
                        href="/details"
                        style="text-decoration: inherit; padding-right: 20px;"
                      >
                         
                        <div
                          class="my-auto"
                        >
                          Carter Robinso 
                        </div>
                      </a>
                    </div>
                  </div>
                </div>
                <div
                  class="my-auto mx-auto text-center py-2 px-1 col-md-6"
                >
                  <img
                    alt="user_picture"
                    class="rounded-circle"
                    height="48px"
                    src="https://randomuser.me/api/portraits/thumb/men/94.jpg"
                    width="48px"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="list-group-item"
          >
            <div
              class="my-3 my-auto flex justify-content-center container"
            >
              <div
                class="row"
              >
                <div
                  class="mx-auto text-center col-md-6"
                >
                  <div
                    class="ms-2 me-auto"
                  >
                    <div
                      class="fw-bold"
                    >
                      <a
                        href="/details"
                        style="text-decoration: inherit; padding-right: 20px;"
                      >
                         
                        <div
                          class="my-auto"
                        >
                          Carter Robinson 
                        </div>
                      </a>
                    </div>
                  </div>
                </div>
                <div
                  class="my-auto mx-auto text-center py-2 px-1 col-md-6"
                >
                  <img
                    alt="user_picture"
                    class="rounded-circle"
                    height="48px"
                    src="https://randomuser.me/api/portraits/thumb/men/94.jpg"
                    width="48px"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="list-group-item"
          >
            <div
              class="my-3 my-auto flex justify-content-center container"
            >
              <div
                class="row"
              >
                <div
                  class="mx-auto text-center col-md-6"
                >
                  <div
                    class="ms-2 me-auto"
                  >
                    <div
                      class="fw-bold"
                    >
                      <a
                        href="/details"
                        style="text-decoration: inherit; padding-right: 20px;"
                      >
                         
                        <div
                          class="my-auto"
                        >
                          Sandyw Scholl 
                        </div>
                      </a>
                    </div>
                  </div>
                </div>
                <div
                  class="my-auto mx-auto text-center py-2 px-1 col-md-6"
                >
                  <img
                    alt="user_picture"
                    class="rounded-circle"
                    height="48px"
                    src="https://randomuser.me/api/portraits/thumb/women/58.jpg"
                    width="48px"
                  />
                </div>
              </div>
            </div>
          </div>
        </ol>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
